### PR TITLE
Add Task, validation for the text field

### PR DESCRIPTION
**Pull request by Bassem0gamal**
Add :
1. add_task_screen
2. add_task_controller
3. Add validation for the  task_screen's Text Fields

### CODE REVIEW

File: lib/data_source/add_task_controller.dart [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import '../task/models/task_model.dart';
+
+class AddTaskController extends GetxController {
+  final TaskDataSource _dataSource = InMemoryTaskDataSource.instance;
+  final RxList<TaskModel> tasks = RxList();
+  bool submitted = false;
+
+  final TextEditingController newTitle = TextEditingController();
+  final TextEditingController newNote = TextEditingController();
+  final TextEditingController newDate = TextEditingController();
```
Comment (ahmedgemi) [inline lib/data_source/add_task_controller.dart]: `GetxController` has a callback function when the controller is disposed. So dispose the TextEditingControllers in this function

File: lib/data_source/add_task_controller.dart [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import 'package:getx_test_1/data_source/add_task_starter.dart';
+import 'package:getx_test_1/data_source/in_memory_task_data_source.dart';
+import 'package:getx_test_1/data_source/task_data_source.dart';
+
+import '../task/models/task_model.dart';
+
+class AddTaskController extends GetxController {
+  final TaskDataSource _dataSource = InMemoryTaskDataSource.instance;
+  final RxList<TaskModel> tasks = RxList();
+  bool submitted = false;
```
Comment (ahmedgemi) [inline lib/data_source/add_task_controller.dart]: Is it used somewhere?

**Comment by Bassem0gamal** [comment]
Add :
1. add_task_screen
2. add_task_controller
3. Add validation for the  task_screen's Text Fields

File: lib/data_source/add_task_starter.dart [hunk 3]
```diff
@@ -0,0 +1,101 @@
+import 'package:flutter/material.dart';
+import 'package:getx_test_1/data_source/add_task_controller.dart';
+import 'package:getx_test_1/icons.dart';
+import 'package:getx_test_1/task/models/task_input_field_model.dart';
+import 'package:getx_test_1/text_style.dart';
+import 'package:intl/intl.dart';
+
+class AddTaskStarter extends StatefulWidget {
+  const AddTaskStarter({Key? key}) : super(key: key);
```
Comment (ahmedgemi) [inline lib/data_source/add_task_starter.dart]: S: add new line after constructor

File: lib/data_source/add_task_controller.dart
Comment (ahmedgemi) [inline lib/data_source/add_task_controller.dart:1]: Move the file to the task package

File: lib/data_source/add_task_starter.dart
Comment (ahmedgemi) [inline lib/data_source/add_task_starter.dart:1]: Move the file to the task package

**Review by ahmedgemi — APPROVED** [review]

**Pull request merged**